### PR TITLE
Add Place Tile hint support in domain API

### PR DESCRIPTION
**Pull request by jonathan-marmelab**
Depends on #46

### CODE REVIEW

File: domain/api/insert_tile.go [hunk 1]
```diff
@@ -36,23 +27,13 @@ func insertTileHandler(w http.ResponseWriter, r *http.Request) {
...
+	err := requestBody.Board.InsertTileAt(requestBody.Direction, requestBody.Index)
+	if err == model.ErrUnsupportedDirection {
 		log.Printf("POST '/insert-tile' - Unsupported direction: %v", requestBody.Direction)
 		http.Error(w, fmt.Sprintf("unsupported direction: %v", requestBody.Direction), http.StatusInternalServerError)
-	}
-
-	actions := make([]*Action, 0, 1)
-	if err == nil {
+		return
+	} else if err == nil {
```
Comment (guilbill) [inline domain/api/insert_tile.go:36]: What happen if the err is neither nil nor ErrUnsupportedDirection?

File: domain/internal/model/hint.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	for _, player := range b.Players {
+		boardCopy.Players = append(boardCopy.Players, player.Copy())
+	}
+
+	copy(boardCopy.RemainingPlayers, b.RemainingPlayers)
+
+	return boardCopy
+}
+
+func (b *Board) GetPlaceTileHint() (*Board, *PlaceTileHint) {
```
Comment (guilbill) [inline domain/internal/model/hint.go:41]: You should test this function.

**Review by guilbill — CHANGES_REQUESTED** [review]
You should add some tests

File: domain/api/insert_tile.go (hunk 1 above)
Comment (jonathan-marmelab) [reply domain/api/insert_tile.go:36]: It can also be `ErrInvalidAction` and `ErrEvenRow` which can be swallowed in the response

**Review by Nitix — APPROVED** [review]

**Pull request merged**